src/components/AppMapDetailsObj.ts

```diff
@@ -415,4 +418,100 @@ export default class AppMapDetailsObj extends AppMapDetailsBase<MapMarkerObj | M
     }
     return "Normal";
   }
+  korokType() {
+    //if (!this.obj || !('korok_id' in this.obj)) {
+    //  return "NaK";
+    //}
+    return this.obj && this.obj.korok_type == 'Flower Trail';
+  }
+
+  findItemByHash(group: any[], links: any[], name: string): any {
+    let hashes = links.map(link => link.DestUnitHashId);
+    let out = group.filter(g => g.data.UnitConfigName == name && hashes.includes(g.hash_id));
+    if (out.length == 1) {
+      return out[0];
+    }
+    return undefined;
```

Comment:
use null, not undefined here (see https://stackoverflow.com/questions/37980559/is-it-better-to-return-undefined-or-null-from-a-javascript-function)